.github/scripts/cmake.sh

```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/env bash
+
+set -euxo pipefail
+
+./.github/scripts/setup-env.sh
+
+# Activate conda environment
+set +x && eval "$($(which conda) shell.bash hook)" && conda deactivate && conda activate ci && set -x
+
+# Setup the OS_TYPE environment variable that should be used for conditions involving the OS below.
+case $(uname) in
+  Linux)
+    OS_TYPE=linux
+    ;;
+  Darwin)
+    OS_TYPE=macos
+    ;;
+  MSYS*)
+    OS_TYPE=windows
+    ;;
+  *)
+    echo "Unknown OS type:" $(uname)
+    exit 1
+    ;;
+esac
+
+if [[ $OS_TYPE == macos ]]; then
+  JOBS=$(sysctl -n hw.logicalcpu)
+else
+  JOBS=$(nproc)
+fi
+
+TORCH_PATH=$(python -c "import pathlib, torch; print(pathlib.Path(torch.__path__[0]))")
+if [[ $OS_TYPE == windows ]]; then
+  PACKAGING_DIR="${PWD}/packaging"
+  export PATH="${TORCH_PATH}/lib:${PATH}"
+fi
+
+Torch_DIR="${TORCH_PATH}/share/cmake/Torch"
+if [[ "${GPU_ARCH_TYPE}" == "cuda" ]]; then
+  WITH_CUDA=1
+else
+  WITH_CUDA=0
+fi
+
+echo '::group::Prepare CMake builds'
+mkdir -p cpp_build
+
+pushd test/tracing/frcnn
+python trace_model.py
+mkdir -p build
+mv fasterrcnn_resnet50_fpn.pt build
+popd
+
+pushd examples/cpp/hello_world
+python trace_model.py
+mkdir -p build
+mv resnet18.pt build
+popd
+
+# This was only needed for the tracing above
+pip uninstall -y torchvision
```

Comment:
Probably not needed, but we are making sure here that we don't need to have a `torchvision` Python install for anything below.